doc/model.rst

```diff
@@ -0,0 +1,719 @@
+
+When specifying a TorchScript using ``Model.add_script()``, the
+following arguments are offered:
+
+- name (str): key to store script under.
+- script (t.Optional[str] = None): TorchScript code (only supported for non-colocated orchestrators).
+- script_path (t.Optional[str] = None): path to TorchScript code.
+- device (t.Literal["CPU", "GPU"] = "CPU"): device for script execution, defaults to “CPU”.
+- devices_per_node (int = 1): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- first_device (int = 0): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+
+You might use TorchScript scripts to represent individual models within the model.
+
+Example: Loading a TorchScript to the Model
+--------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an TorchScript into the database at model runtime.
+
+**Python Script: Using Torch Scripts As ML Models**
+
+Define the TorchScript code as a variable in the Python driver script:
+
+.. code-block:: python
+
+    torch_script_str = "def negate(x):\n\treturn torch.neg(x)\n"
+
+**SmartSim Model Integration:**
+
+Assuming an initialized ``Model`` named `smartsim_model` exists, we add a TensorFlow model using
+the ``Model.add_script()`` function:
+
+.. code-block:: python
+
+    smartsim_model.add_script(name="example_script", script=torch_script_str, device="GPU", devices_per_node=2, first_device=0)
+
+In the above ``smartsim_model.add_script()`` code snippet, we offer the following arguments:
+
+-  name ("example_script"): key to store script under.
+-  script (torch_script_str): TorchScript code.
+-  device ("CPU"): device for script execution.
+-  devices_per_node (2): Use two GPUs per node.
+-  first_device (0): Start with 0 index GPU.
+
+When the model is started via ``Experiment.start()``, the TorchScript will be loaded to the
+orchestrator that is launched prior to the start of the model.
+
+=========================
+Data Collision Prevention
+=========================
+--------
+Overview
+--------
+SmartSim and SmartRedis can avoid key collision by prepending program-unique
+prefixes to Model workloads launched through SmartSim. For example, if you were
+to have two applications feeding data to a single database, who produced keys
+of the same name, upon requesting this information there would be a key collision
+since there is no yet uniqueness between the same tensor names. By enabling key
+prefixing on a Model, SmartSim will append the model name to each key produced
+by the application and sent to the database as such: `model_name.tensor_name`.
+
+This is done simply through functions offered by the Model API:
+
+* Model.register_incoming_entity(incoming_entity)
+* Model.enable_key_prefixing()
+* Model.disable_key_prefixing()
+* Model.query_key_prefixing()
+
```

Comment:
I think we need to do a thorough overview of the default behvior of each data type with key prefixing (tensor, model, script, dataset, aggregation list) on both send and receives.  Users easily get tripped on this.  I think we could maybe put this into a nice table that is in a separate file so it can be used in multiple locations.